src/imitation/algorithms/adversarial.py

```diff
@@ -411,73 +411,71 @@ def _build_disc_feed_dict(
-        max_episode_steps=max_episode_steps,
-    )
-    gen_policy = util.init_rl(env, verbose=1, **init_rl_kwargs)
-
-    if use_gail:
+class GAIL(AdversarialTrainer):
+    def __init__(
+        self,
+        venv: VecEnv,
+        expert_demos: types.Transitions,
+        gen_policy: base_class.BaseRLModel,
+        *,
+        discrim_kwargs: Optional[Mapping] = None,
+        **kwargs,
+    ):
+        """Generative Adversarial Imitation Learning.
+
+        Most parameters are described in and passed to `AdversarialTrainer.__init__`.
+        Additional parameters that `GAIL` adds on top of its superclass initializer are
+        as follows:
+
+        Args:
+            discrim_kwargs: Optional keyword arguments to use while constructing the
+                DiscrimNetGAIL.
+
+        """
+        discrim_kwargs = discrim_kwargs or {}
         discrim = discrim_net.DiscrimNetGAIL(
-            env.observation_space, env.action_space, scale=scale, **discrim_kwargs
-        )
-    else:
-        rn = BasicShapedRewardNet(
-            env.observation_space, env.action_space, scale=scale, **reward_kwargs
+            venv.observation_space, venv.action_space, **discrim_kwargs
         )
-        discrim = discrim_net.DiscrimNetAIRL(
-            rn, entropy_weight=airl_entropy_weight, **discrim_kwargs
+        super().__init__(venv, gen_policy, discrim, expert_demos, **kwargs)
+
+
+class AIRL(AdversarialTrainer):
+    def __init__(
+        self,
+        venv: VecEnv,
+        expert_demos: types.Transitions,
+        gen_policy: base_class.BaseRLModel,
+        *,
+        reward_net_cls: Type[reward_net.RewardNet] = reward_net.BasicShapedRewardNet,
+        reward_net_kwargs: Optional[Mapping] = None,
+        discrim_kwargs: Optional[Mapping] = None,
+        **kwargs,
+    ):
+        """Adversarial Inverse Reinforcement Learning.
+
+        Most parameters are described in and passed to `AdversarialTrainer.__init__`.
+        Additional parameters that `AIRL` adds on top of its superclass initializer are
+        as follows:
+
+        Args:
+            reward_net_cls: Reward network constructor. The reward network is part of
+                the AIRL discriminator.
+            reward_net_kwargs: Optional keyword arguments to use while constructing
+                the reward network.
+            discrim_kwargs: Optional keyword arguments to use while constructing the
+                DiscrimNetAIRL.
+        """
+        # TODO(shwang): Maybe offer str=>Type[RewardNet] conversion like
+        #  stable_baselines does with policy classes.
+        reward_net_kwargs = reward_net_kwargs or {}
+        reward_network = reward_net_cls(
+            action_space=venv.action_space,
+            observation_space=venv.observation_space,
+            **reward_net_kwargs,  # pytype: disable=not-instantiable
         )
+        # pytype is afraid that we'll directly call RewardNet() which is an abstract
+        # class, hence the disable.
```

Comment:
I expect comments to be before what happens, so either say `disable above` or move this to be before.